Utils/Dataflow/017_adjustMetadata/adjustMetadata.py

```diff
@@ -206,7 +200,13 @@ def transform_chain_data(data, original):
 def process(stage, message):
     """ Single message processing. """
     data = message.content()
-    original = str(data)
+
+    if type(data) is not dict:
+        sys.stderr.write('(WARN) Message contains non-dict data: %r. '
+                         'Skipping.\n' % data)
+        return False
+    if 'taskid' not in data:
+        sys.stderr.write('(WARN) Message contains no taskid: %r.\n' % data)
```

Comment:
Are messages without `taskid` allowed for further processing?

Some steps may be executed correctly (maybe even all of them), but if this stage is to work with messages with task metadata -- it would be fair, I believe, to say "the input message doesn\`t seem to contain task metadata, won\`t even try to process it" or "...processing may fail". But anyway I think it would be better to do something about it, to explain what does the absence of `taskid` mean for this stage.

Reply:
Messages without taskid are skipped now (in another branch, see above).